Tests/SwiftParserTest/Assertions.swift

```diff
@@ -524,6 +524,39 @@ extension ParserTestCase {
     }
   }
 
+  enum FixItsApplication {
+    /// Apply only the fix-its whose messages are included in `applyFixIts` to generate `fixedSource`.
+    case optIn(applyFixIts: [String], fixedSource: String)
+    /// Apply all fix-its to generate `fixedSource`.
+    case all(fixedSource: String)
+
+    init?(applyFixIts: [String]?, expectedFixedSource: String?) {
+      if let applyFixIts {
+        self = .optIn(applyFixIts: applyFixIts, fixedSource: expectedFixedSource ?? "")
+      } else if let expectedFixedSource {
+        self = .all(fixedSource: expectedFixedSource)
+      } else {
+        return nil
+      }
+    }
+
+    var applyFixIts: [String]? {
+      switch self {
+      case .optIn(let applyFixIts, _):
+        applyFixIts
+      case .all:
+        nil
+      }
+    }
+
+    var expectedFixedSource: String {
+      switch self {
+      case .optIn(_, let fixedSource), .all(let fixedSource):
+        fixedSource
+      }
+    }
```

Comment:
swift-syntax needs to be able to build with Swift 5.8, so you can’t use switch expressions here.

```suggestion
    var applyFixIts: [String]? {
      switch self {
      case .optIn(let applyFixIts, _):
        return applyFixIts
      case .all:
        return nil
      }
    }

    var expectedFixedSource: String {
      switch self {
      case .optIn(_, let fixedSource), .all(let fixedSource):
        return fixedSource
      }
    }
```